builtin/fstk/ui.lua

```diff
@@ -60,11 +60,32 @@ function ui.update()
 
 	-- handle errors
 	if gamedata ~= nil and gamedata.errormessage ~= nil then
-		formspec = "size[12,3.2]" ..
-			"textarea[1,1;10,2;;ERROR: " ..
-			core.formspec_escape(gamedata.errormessage) ..
-			";]"..
-			"button[4.5,2.5;3,0.5;btn_error_confirm;" .. fgettext("Ok") .. "]"
+		formspec = ""
+		local ar = gamedata.errormessage:split("\n")
+		for i = 1, #ar do
+			local text = ar[i]
+			while #text > 80 do
+				if formspec ~= "" then
+					formspec = formspec .. ","
+				end
+				formspec = formspec .. core.formspec_escape(string.sub(text, 1, 79))
+				text = string.sub(text, 80, #text)
+			end
+			if formspec ~= "" then
+				formspec = formspec .. ","
+			end
+			formspec = formspec .. core.formspec_escape(text)
+		end
+		local error_title = ""
+		if string.find(gamedata.errormessage, "ModError") then
```

Comment:
I don't like string searches to handle logic like this, but at least use a raw match instead of a regex match.